### ISSUE TITLE
Come up with CSS for reseting quicksearch

### ISSUE THREAD

**Issue by romaninsh** [atk4/ui#365]
http://ui.agiletoolkit.org/demos/grid.php has a QuickSearch bar. When user types something and click on the search icon, grid reflects that. If previous search exists user would have to delete text manually if he wanted to reset search.

We need to improve usability of QuickSearch by allowing:

 - perform search without pressing enter (button)
 - reset search without touching keyboard

## Concept 1: JS action substitution

Using Action buttons show "reset" button by default after successful search. If user clicks into the field (usually with intent to change), then hide reset and show search action.

![quicksearch](https://user-images.githubusercontent.com/453929/36026638-e8419cc6-0d8f-11e8-89a8-2d97f8604467.gif)

``` php
$qs=$app->add('FormField\Input');
$act_search = $qs->add(['Button', 'icon'=>'search'], 'AfterInput')->addStyle('display', 'none');
$act_reset = $qs->add(['Button', 'icon'=>'red cancel'], 'AfterInput');
$qs->addClass('action');
$qs->set('prevsearch');

$act_search->on('click', $qs->jsInput()->val('searching'));
$act_reset->on('click', $qs->jsInput()->val('reset'));

$qs->on('focus', 'input', new \atk4\ui\jsExpression('[];[]',[$act_search->js()->show(), $act_reset->js()->hide()]));
```

## Concept 2: Corner reset

Use less-intrusive corner label to display cancel, while still providing search action:

![quicksearch2](https://user-images.githubusercontent.com/453929/36026903-ea75d4de-0d90-11e8-9c14-ee8ff1d25169.gif)

``` php
$i=$app->add('FormField\Input');
$i->addAction(['icon'=>'search']);
$label = $i->add(['Label','', 'left red corner'], 'AfterInput')->addStyle('cursor', 'pointer');
$label->add(['Icon','cancel']);
$i->addClass('left corner labeled');
$i->on('click', '.label', $label->js()->hide());
$i->on('click', '.label', $i->js()->removeClass('left corner labeled'));
$i->on('click', '.label', $i->jsInput()->val('')->focus());
$i->set('prevsearch');
```

# Problem:

Neither concept actually works when placed in Grid. Input field is located inside Menu with class "transparent" to avoid extra spacing. Labels appear miss-aligned:

<img width="232" alt="screen shot 2018-02-09 at 12 03 10" src="https://user-images.githubusercontent.com/453929/36026979-39e5c696-0d91-11e8-8586-5b625767723d.png">

and action adds extra padding around:

<img width="270" alt="screen shot 2018-02-09 at 12 02 31" src="https://user-images.githubusercontent.com/453929/36026958-2a3ae26c-0d91-11e8-84c4-cf3030e024ed.png">

Neither of them is acceptable. Either some CSS tweaking is needed or some alternative solution is required.

**Final note**: solution with changing icon should be also OK, but I can't catch click event on the icon as it seems to be behind the field.

**Comment by DarkSide666** [comment]
one more option is to add 2 buttons in input field if that's possible

**Comment by romaninsh** [comment]
as mentioned before - buttons are borken.

**Comment by romaninsh** [comment]
Currently #371 implement it like this:

<img width="340" alt="screen shot 2018-02-14 at 11 11 57" src="https://user-images.githubusercontent.com/453929/36196000-fd0e299e-1177-11e8-90ae-74b3ab7a56f5.png">

It is still not visually great.

**Comment by romaninsh** [comment]
solved by #375

**Comment by acicovic** [comment]
As stated this has been solved by #371, #375 and #370 so I'm closing this.
